### PR TITLE
Support .labrc.js for configuration

### LINKED ISSUE #598: Support .labrc files for configuration defaults

### PR DESCRIPTION
**Pull request by geek**
Closes #598

### CODE REVIEW

File: test/run_cli.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ const internals = {
 
 module.exports = (args, callback, root) => {
 
-    const cli = ChildProcess.spawn('node', [].concat(internals.labPath, args), { 'cwd' : root ? root : '.' });
+    const cli = ChildProcess.spawn('node', [].concat(internals.labPath, args), { cwd : root ? root : '.' });
```
Comment (cjihrig) [inline test/run_cli.js]: While you're here... `root ? root : '.'` could just be `root || '.'` I think.

File: test/cli_labrc/.labrc [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
```
Comment (cjihrig) [inline test/cli_labrc/.labrc]: Does the find rc module support file extensions? Being able to specify .js would be nice at least for syntax highlighting.

File: README.md [hunk 3]
```diff
@@ -367,6 +367,26 @@ if (typeof value === 'symbol') {
 
 ```
 
+## `.labrc` file
+
+**lab** supports a `.labrc` configuration file for centralizing lab settings.  
+The `.labrc` file can be located in the current working directory or in the
+users home directory.  The `.labrc` file needs to be able to be required by
```
Comment (cjihrig) [inline README.md]: user -> user's

Comment (cjihrig) [inline README.md]: The second sentence might start better as "The .labrc file must be requirable by"

**Comment by cjihrig** [comment]
Maybe I'm missing something, but where do I look to see what options are supported by this new file? Are the config options just the name of the long command line flags without the `--` in front?

File: test/cli_labrc/.labrc (hunk 2 above)
Comment (gergoerdosi) [inline test/cli_labrc/.labrc]: I agree, would be good to use an extension. Eslint deprecated `.eslintrc` too in favor of `.eslintrc.json`, `.eslintrc.yaml `, etc.

**Comment by geek** [comment]
I will update the readme to make it clearer what the allowed options are.  They are the same long names as the cli options.

**Comment by Marsup** [comment]
Do you think we should be able to provide our own labrc or do you prefer a base labrc overridden through cli ?

**Comment by geek** [comment]
@cjihrig should be ready to go.  I would rather the file be `.labrc` instead of supporting `.labrc.json` and `.lanrc.js`.  

@Marsup you can provide your own labrc, lab doesn't ship with one.

**Comment by Marsup** [comment]
@geek I didn't mean that. You always have multiple test configuration, how do you intend to deal with that + rc ?

**Comment by geek** [comment]
@Marsup the first one found closest to the heart of the project wins.

1. Current directory
2. Parent of current directory, until the root folder is encountered
3. $HOME/.${apprc}

**Comment by cjihrig** [comment]
LGTM

**Comment by yelworc** [comment]
Quick note: The removed default for `NODE_ENV` caught me by surprise – my tests were suddenly running with the wrong app configuration after updating lab...

**Comment by geek** [comment]
@yelworc thanks for the report.  I will investigate the issue, it shouldn't have happened because there is a default environment set in the runner https://github.com/hapijs/lab/blob/master/lib/runner.js#L40

**Pull request merged**